zebra-state/src/service/finalized_state/disk_format/transparent.rs

```diff
@@ -498,14 +498,17 @@ impl AddressTransaction {
 
 /// Returns a byte representing the [`transparent::Address`] variant.
 fn address_variant(address: &transparent::Address) -> u8 {
+    use NetworkKind::*;
     // Return smaller values for more common variants.
     //
     // (This probably doesn't matter, but it might help slightly with data compression.)
     match (address.network(), address) {
         (Mainnet, PayToPublicKeyHash { .. }) => 0,
         (Mainnet, PayToScriptHash { .. }) => 1,
-        (Testnet, PayToPublicKeyHash { .. }) => 2,
-        (Testnet, PayToScriptHash { .. }) => 3,
+        // There's no way to distinguish between Regtest and Testnet for encoded transparent addresses,
+        // so the network kind should always be `Mainnet` or `Testnet`.
```

Comment:
Shouldn't the following line say:

```suggestion
        // so the network kind should always be `Testnet` or `Regtest` for network kinds other than Mainnet.
```
?

Reply:
Yeah, I meant to convey something like "Regtest uses Testnet transparent addresses", I'm happy to go with your suggestion, or a slightly more verbose:

```suggestion
        // we can consider `Regtest` to use `Testnet` transparent addresses, so it's okay to use the `Testnet`
        // address variant for `Regtest` transparent addresses in the db format
```

Reply:
Go ahead with yours, it makes more sense.